dev/ci/ci-user-overlay.sh

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+
+# Add user overlays here. You can use some logic to detect if you are
+# in your travis branch and conditionally enable the overlay.
```

Comment:
What is the problem with dropping commits? My feeling is that the standard workflow is going to be:
- Create a PR
- Travis detects some breaking changes
- Create branches for the affected developments which fix these changes in a backward compatible way (whenever possible) and submit them as PRs, or ask someone to do that for you
- Add a special commit adding overlays to test your branch before these other PRs get merged
- Wait for your other PRs to get merged
- Drop the commit with the overlays since it's not needed anymore
- Merge the initial PR

Not dropping the overlay commit means cluttering the history. And custom Travis logic creates a risk that the PR was passing but that it will break once in trunk.